src/scraper/scraper.cpp

```diff
@@ -264,6 +281,40 @@ class logger
 
             fs::remove(pfile_temp);
 
+            bool fDeleteOldLogArchives = GetBoolArg("-deleteoldlogarchives", true);
+
+            if (fDeleteOldLogArchives)
+            {
+                unsigned int nRetention = (unsigned int)GetArg("-logarchiveretainnumfiles", 14);
+
+                std::set<fs::directory_entry, std::greater <fs::directory_entry>> SortedDirEntries;
+
+                // Iterate through the log archive directory and delete the oldest files beyond the retention rule
+                // The names are in format scraper-YYYYMMDDHHMMSS for the scraper logs, so filter by containing scraper
+                // The greater than sort in the set should then return descending order by datetime.
+                for (fs::directory_entry& DirEntry : fs::directory_iterator(LogArchiveDir))
+                {
+                    std::string sFilename = DirEntry.path().filename().string();
+                    size_t FoundPos = sFilename.find("scraper");
+
+                    if (FoundPos != string::npos) SortedDirEntries.insert(DirEntry);
+                }
+
+                // Now iterate through set of filtered filenames. Delete all files greater than retention count.
+                unsigned int i;
```

Comment:
Since I've been learning C++, I realized that primitives don't necessarily initialize to any defined value--bit me a couple times. It seems like I got lucky that `i` initialized to `0` when I ran the `archivescraperlog` command the first time. On the subsequent run, the variable contained whatever value in memory at that address, which was greater than `nRetention`, so it erased the remaining items.

Reply:
Yep. In my haste I forgot to initialize the i to 0.

Reply:
What is even worse is that some things are initialized to empty, etc. even when not specified and some things are not...

Reply:
Looks like the Windows issue is the damn file handle/rename problem. The reason the midnight version works, is that is the instance that is holding the handle...

Reply:
The i = 0 isn't it... WTF.

Reply:
@cyrossignol If you want to poke around and see if you can figure out why the windows version calling the archivescraperlog doesn't work, that would be a big help. I have to head to NYC now. The midnight auto archive has been working... I also tested RC2 (before these changes) on Windows with archivescraperlog, and it fails too (returns false).

Reply:
I'll play with it a bit when I get home...don't have a Windows box to try it on at the moment.